### PR TITLE
Adds the Vendor view

### PR DESCRIPTION
**Pull request by Huntrr**
Vendors can add and manage their items.

### CODE REVIEW

File: app/vendor/views.py [hunk 1]
```diff
@@ -1,11 +1,115 @@
...
+    return render_template('vendor/index.html')
+
+@vendor.route('/new-item', methods=['GET', 'POST'])
+@login_required
+@vendor_required
+def new_listing():
+    """Create a new item."""
+    form = NewItemForm()
+    if form.validate_on_submit():
+        category_id = form.categoryId.data.name
```
Comment (jondubin) [inline app/vendor/views.py]: The category id is getting set to the category name

File: app/vendor/forms.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	listingPrice = IntegerField('Item Price', validators=[DataRequired()])
+	listingAvailable = BooleanField('Available?')
+	submit = SubmitField('Update Item Information')
+
+class NewItemForm(Form):	
+	categoryId = QuerySelectField('Category', validators=[DataRequired()], get_label='name', query_factory=lambda: db.session.query(Category).order_by('id'))
+	listingName = StringField('Item Name', validators=[DataRequired(), Length(1, 1000)])
+	listingDescription = TextAreaField('Item Description', validators=[DataRequired(), Length(1, 2500)])
+	listingPrice = IntegerField('Item Price', validators=[DataRequired()])
+	listingAvailable = BooleanField('Available?')
```
Comment (jondubin) [inline app/vendor/forms.py]: Can we make the default value for this true/checked?

File: app/admin/views.py [hunk 3]
```diff
@@ -102,7 +102,6 @@ def delete_category(category_id):
     return redirect(url_for('admin.view_categories'))
 
 
-
```
Comment (Huntrr) [inline app/admin/views.py:106]: Is this third line break a PEP8 thing? i.e. was it here on purpose?

File: app/vendor/views.py [hunk 4]
```diff
@@ -1,11 +1,116 @@
...
+    form.categoryId.default = listing.category
+    form.listingAvailable.default = listing.available
+    form.process()
+    return render_template('vendor/manage_listing.html', listing=listing, form=form)
+
+
+@vendor.route('/item/<int:listing_id>/delete')
+@login_required
+@vendor_required
+def delete_listing_request(listing_id):
```
Comment (Huntrr) [inline app/vendor/views.py:103]: Yeah, um, so this route lets ANY vendor delete ANY item.

That's not good.

File: app/vendor/views.py [hunk 5]
```diff
@@ -1,11 +1,116 @@
...
+    return render_template('vendor/current_listings.html',
+                           listings=listings,
+                           categories=categories)
+
+
+@vendor.route('/items/<int:listing_id>')
+@vendor.route('/items/<int:listing_id>/info')
+@login_required
+@vendor_required
+def listing_info(listing_id):
```
Comment (Huntrr) [inline app/vendor/views.py:62]: Same issue as `delete_listing`. Vendors shouldn't be able to view other vendors listings by just entering a url.

File: app/vendor/views.py [hunk 6]
```diff
@@ -1,11 +1,116 @@
...
+    listing = Listing.query.filter_by(id=listing_id).first()
+    if listing is None:
+        abort(404)
+    return render_template('vendor/manage_listing.html', listing=listing)
+
+
+@vendor.route('/items/<int:listing_id>/edit-item', methods=['GET', 'POST'])
+@login_required
+@vendor_required
+def change_listing_info(listing_id):
```
Comment (Huntrr) [inline app/vendor/views.py:75]: Same as `delete_listing`. Plz check `listing.vendor_id == current_user.id` (`current_user is listing.vendor` might also work and be more idiomatic, I'm not sure).

File: app/vendor/views.py [hunk 7]
```diff
@@ -1,11 +1,116 @@
...
+    """Change a listings's info."""
+    listing = Listing.query.filter_by(id=listing_id).first()
+    if listing is None:
+        abort(404)
+    form = ChangeListingInformation()
+    if form.validate_on_submit():
+        listing.category_id,listing.name, listing.description, listing.available, listing.price, listing.vendor_id = form.categoryId.data.id, form.listingName.data, form.listingDescription.data, form.listingAvailable.data, form.listingPrice.data, current_user.id
```
Comment (jondubin) [inline app/vendor/views.py]: Split this line up

**Comment by jondubin** [comment]
:heart:

File: app/admin/views.py (hunk 3 above)
Comment (jondubin) [inline app/admin/views.py:106]: Two lines b/w function definitions for PEP8

**Comment by jondubin** [comment]
@Huntrr fixed

**Pull request merged**